lib/src/Uncertainty/Algorithm/MetaModel/FunctionalChaos/FunctionalChaosResult.cxx

```diff
@@ -271,6 +277,138 @@ Function FunctionalChaosResult::getComposedMetaModel() const
   return composedMetaModel_;
 }
 
+/* Conditional expectation accessor */
+FunctionalChaosResult FunctionalChaosResult::getConditionalExpectation(const Indices & conditioningIndices) const
+{
+  // Compute active dimension
+  const UnsignedInteger inputDimension = inputSample_.getDimension();
+  const UnsignedInteger activeDimension = conditioningIndices.getSize();
+  for (UnsignedInteger conditioningIndex = 0; conditioningIndex < activeDimension; ++ conditioningIndex)
+  {
+    const UnsignedInteger variableIndex = conditioningIndices[conditioningIndex];
+    if (variableIndex >= inputDimension)
+      throw InvalidArgumentException(HERE) << "Active indice" << variableIndex 
+        << "in conditioningIndices is not consistent with input dimension"
+        << inputDimension;
+  }
```

Comment:
this block can be replaced with the condition: "conditioningIndices.check(inputDimension)"

Reply:
Thank you for the tip. It even improves the check. Ok, fixed at d25ac944b64f87beb412d25ccbf066a492ee8c1f.